### ISSUE TITLE
when the page number of the pdf file is bigger then 300,  the extract_pages(f) is very slow

### ISSUE THREAD

**Issue by tianyongliu** [pdfminer/pdfminer.six#1071]
#
[中国神华：中国神华2023年度报告.PDF](https://github.com/user-attachments/files/18120514/2023.PDF)

code is very simple 

from pdfminer.high_level import extract_pages
file_path = "***.pdf"
with open (file_path, "rb") as f:
  for page in extract_pages(f):
    page_index = page.pageid
    print(page_index)

it cost about 10 minutes.

**Comment by dhdaines** [comment]
Well, `extract_pages` is actually doing layout analysis on every page, so one might expect this to be slow!  Do you find that it gets slower with each successive page?

**Comment by tianyongliu** [comment]
yes! It is fast with a smaller file and will be slower with a bigger file!
It is possible to process it with multiprocess ?

**Comment by TheCutestCat** [comment]
@tianyongliu  YES, we can process with multiprocess, check it out there : [link](https://github.com/dhdaines/playa/issues/32)

**Comment by pietermarsman** [comment]
Can't reproduce this with the current version of pdfminer.six. Can you try that? 

Extracting all pages takes 27 seconds. 

```
$ time .venv/bin/python tools/pdf2txt.py ~/Downloads/2023.PDF
[ lots of output ]
.venv/bin/python tools/pdf2txt.py ~/Downloads/2023.PDF  26.92s user 0.10s system 99% cpu 27.146 total
```

Extracting the first 150 pages takes 10 seconds.

```
$ time .venv/bin/python tools/pdf2txt.py ~/Downloads/2023.PDF --page-numbers $(python -c 'print(" ".join(map(str, range(150))))')
[ lots of output ]
.venv/bin/python tools/pdf2txt.py ~/Downloads/2023.PDF --page-numbers   9.70s user 0.06s system 99% cpu 9.811 total
```

Extracting the last 150 pages takes 16 seconds.

```
$ time .venv/bin/python tools/pdf2txt.py ~/Downloads/2023.PDF --page-numbers $(python -c 'print(" ".join(map(str, range(150, 307))))')
[ lots of output ]
.venv/bin/python tools/pdf2txt.py ~/Downloads/2023.PDF --page-numbers   16.25s user 0.09s system 99% cpu 16.406 total
```

So extracting all pages is the same time as extracting only the first half + extracting only the second half. No effect of the number of pages that are extracted. 

Closing this, but feel free to comment if you observe differently.
